orttraining/orttraining/python/training/utils/hooks/_zero_offload_subscriber.py

```diff
@@ -55,6 +58,49 @@ def _setup_zero_stage3_ort_compatible_hooks(self):
 
     return _setup_zero_stage3_ort_compatible_hooks
 
+class DummyWorkZeroStage3(torch.distributed.distributed_c10d.Work):
+    # def __init__(self):
+    #     pass
+
+    def is_completed(self) -> bool:
+        return True
+    def is_success(self) -> bool:
+        return True
+    def exception(self) -> Any:
+        return None
+    def wait(self, timeout: timedelta = timedelta) -> bool:
+        return True
+    def source_rank(self) -> int:
+        return 0
+    def _source_rank(self) -> int:
+        return 0
+    def result(self) -> List[torch.Tensor]:
+        return []
+    def synchronize(self):
+        pass
+
+
+def _ort_compatible_allgather_fn():
+    # For Monkey patching the original function
+    # Original code https://github.com/microsoft/DeepSpeed/blob/604d701e35548e5407b017c088bdc3760832c9e0/deepspeed/comm/comm.py#L315
+    def _ort_compatible_allgather_fn_zero_stage3(output_tensor, input_tensor, group=None, async_op=False, debug=get_caller_func()):
```

Comment:
a better way can be this. 

original all_gather_func:

`
def all_gahter_fn(...):
 ...impl...
`
```
def wrapped_all_gather_fn(...):
        if torch.onnx.is_in_onnx_export():
            return DummyWorkZeroStage3()

    all_gahter_fn()
```